src/query/shorthand.ts

```diff
@@ -91,6 +102,45 @@ export function spec(specQ: SpecQuery,
   return parts.join('|');
 }
 
+export function filter(_filter: string | Filter | (string | Filter)[]): string {
+  let filterExpression = '';
+
+  if (_filter instanceof Array) {
+    if (_filter.length > 0) {
+      if (typeof _filter[0] === 'string') {
```

Comment:
Your logic here doesn't allow mixing up `string`  and object. 
Why don't you do just a simple map and check type inside the map and just join at the end. 

Plus, I think `) && (` is a better delimiter with `(` and `)` at the beginning and end.